### PR TITLE
OCPBUGS-30774: always save serial logs if they were gathered 

### PR DESCRIPTION
**Pull request by r4f4**
When the installer gathers a log bundle after failure (either automatically or with gather bootstrap), the installer fails to return serial console logs if an SSH connection to the bootstrap node is refused.

Instead let's always save the console serial logs if they were collected.

### CODE REVIEW

**Comment by r4f4** [comment]
`altinfra-e2e-aws-ovn`: shows that when `ssh` succeeds, the console logs are still saved, as expected: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_installer/8263/pull-ci-openshift-installer-master-altinfra-e2e-aws-ovn/1778780388383002624/artifacts/e2e-aws-ovn/ipi-install-install/artifacts/

**Comment by r4f4** [comment]
`e2e-aws-ovn`: ssh connection to bootstrap failed but serial logs collected: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_installer/8263/pull-ci-openshift-installer-master-e2e-aws-ovn/1778776351092772864/artifacts/e2e-aws-ovn/ipi-install-install/artifacts/

**Comment by r4f4** [comment]
Example of failure before the fix: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8264/pull-ci-openshift-installer-master-e2e-aws-ovn/1778790262227603456

**Comment by r4f4** [comment]
See https://github.com/openshift/installer/pull/8264 for the ssh connection fix.

**Comment by r4f4** [comment]
Update: commit forcing a bootkube failure has been removed.

/hold cancel

File: cmd/openshift-install/gather.go [hunk 1]
```diff
@@ -154,9 +155,32 @@ func gatherBootstrap(bootstrap string, port int, masters []string, directory str
...
+	}
+
+	if len(archives) == 0 {
+		return "", fmt.Errorf("failed to gather VM console and bootstrap logs")
+	}
+
+	logBundlePath := filepath.Join(directory, fmt.Sprintf("log-bundle-%s.tar.gz", gatherID))
+	err = serialgather.CombineArchives(logBundlePath, archives)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to combine archives")
```
Comment (barbacbd) [inline cmd/openshift-install/gather.go:177]:
```suggestion
		return "",fmt.Errorf("failed to combine archives: %w", err)
```

Comment (r4f4) [reply cmd/openshift-install/gather.go:177]: Because I want to backport this, I thought about replacing the deprecated errors lib in a follow-up PR.

**Pull request merged**